prototype_source/backend_config_tutorial.rst

```diff
@@ -0,0 +1,400 @@
+then we will quantize the given pattern using this DTypeConfig. Otherwise,
+the QConfig is ignored and the pattern will not be quantized.
+
+There are two ways of specifying ``input_dtype``, ``output_dtype``, and
+``weight_dtype``, as simple ``torch.dtype`` or as
+``DTypeWithConstraints``. The constraints currently supported are:
+
+- **quant_min_lower_bound** and **quant_max_upper_bound**: Lower and upper
+  bounds for the minimum and maximum quantized values respectively. If the
+  QConfig’s ``quant_min`` and ``quant_max`` fall outside this range, then
+  the QConfig will be ignored.
+- **scale_min_lower_bound** and **scale_max_upper_bound**: Lower and
+  upper bounds for the minimum and  aximum scale values respectively. If
+  the QConfig’s minimum scale value (currently exposed as ``eps``) falls
+  below the lower bound, then the QConfig will be ignored. Note that the
+  upper bound is currently not enforced.
+- **scale_exact_match** and **zero_point_exact_match**: Exact match
+  requirements for scale and zero point, to be used for operators with
+  fixed quantization parameters such as sigmoid and tanh. If the observer
+  specified in the QConfig is neither ``FixedQParamsObserver`` nor
+  ``FixedQParamsFakeQuantize``, or if the quantization parameters don't
+  match, then the QConfig will be ignored.
+
+End-to-End Example
+------------------
+
+Here we define a simple example BackendConfig that supports 3 things:
+
+- Quantizing linear to int8 (unsigned)
+- Fusing conv2d followed by relu
+- Quantizing the fused conv2d-relu to int8 (unsigned)
+
+In this section, we will run through an example model with these
+patterns and show how to use this BackendConfig in ``prepare_fx`` and
+``convert_fx``.
+
+.. code:: ipython3
+
+    import torch
+    from torch.ao.quantization import (
+        default_weight_observer,
+        get_default_qconfig_mapping,
+        MinMaxObserver,
+        QConfig,
+        QConfigMapping,
+    )
+    from torch.ao.quantization.backend_config import (
+        BackendConfig,
+        BackendPatternConfig,
+        DTypeConfig,
+        DTypeWithConstraints,
+        ObservationType,
+    )
+    from torch.ao.quantization.quantize_fx import prepare_fx, convert_fx
+
+.. code:: ipython3
+
+    # ======================
+    #  Custom BackendConfig
+    # ======================
+    
+    quint8_with_constraints = DTypeWithConstraints(
+        dtype=torch.quint8,
+        quant_min_lower_bound=0,
+        quant_max_upper_bound=255,
+        scale_min_lower_bound=2 ** -12,
+    )
+    
+    weighted_int8_dtype_config = DTypeConfig(
+        input_dtype=quint8_with_constraints,
+        output_dtype=quint8_with_constraints,
+        weight_dtype=torch.qint8,
+        bias_dtype=torch.float)
+
+   def fuse_conv2d_relu(is_qat, conv, relu):
+       """Return a fused ConvReLU2d from individual conv and relu modules."""
+       return torch.ao.nn.intrinsic.ConvReLU2d(conv, relu)
+    
+    # For quantizing Linear
```

Comment:
I think we can add a bit more explanation here about what this config means exactly, we have docs before for each field, but explaining them in a real example might be more helpful to users I think, same for other configs we have

Reply:
especially for observation_type and dtype_config, since these two things we might reuse in the new flow as well.

please let me know if you need some clarification for dtype_config, I think that one is a bit tricky to understand